### DISCUSSION TITLE
Mounting to Starlette app adds trailing '/'

### DISCUSSION THREAD

**Discussion by pattybardo** [encode/starlette#2548] [Q&A] [answered]
I am running a FastAPI server, and mounted a asgi middleware to be able to serve prometheus metrics on the same server at the `/metrics` endpoint. However, when mounting like this, and accessing at said endpoint, a 307 Temporary Redirect is received which redirects to `/metrics/`. This is okay when viewing through the browser, however when querying via the API, the redirect cannot be followed. 

Now the scrape endpoint can be changed to have a trailing slash, but it involves custom changes to the scrape targets, and via most defaults, I would prefer this to be on the '/metrics' endpoint. I think I found the code section that causes this issue for me [routing.py](https://github.com/encode/starlette/blob/master/starlette/routing.py#L411C25-L411C39). What do you think? Am I doing something wrong? Or is is a fair use case for this to be configurable?

My violating code block:
``` python
...
# Add prometheus asgi middleware to route /metrics requests
metrics_app = make_asgi_app()
app.mount(path="/metrics", app=metrics_app)
...
```

**Comment by hasansezertasan** [comment, marked as answer]
As far as I know, that's the default behavior, not a problem but feature.

**Comment by thisbra** [comment]
This is annoying because it pollutes the logs with constants `GET /metrics/ HTTP/1.1' 307 Temporary Redirect`

As a turnaround I've defined the `/metrics` route on FastAPI and used prometheus_client's `generate_latest()` method.

```python
@app.get("/metrics")
async def metrics_endpoint():
    # Generate the Prometheus metrics and return them as a plain text response
    metrics_data = prometheus_client.generate_latest()
    return Response(content=metrics_data, media_type="text/plain")
```

**Comment by sirianni** [comment]
See https://github.com/encode/starlette/issues/869 for more discussion
